### PR TITLE
Revert revert of Micronaut support for airbyte-server

### PR DESCRIPTION
**Pull request by jdpgrailsdev**
## What
* Revert [the revert of the Micronaut changes to airbyte-server](https://github.com/airbytehq/airbyte/pull/23109)
* Offload slow/Temporal-related endpoints to avoid blocking
* Copy of https://github.com/airbytehq/airbyte/pull/23148

## How
* Revert the previous revert
* Mark endpoints that interact with Temporal as candidates for offloading from the Netty event loop

## Recommended reading order
1. `*Controller.java`
2. `application.yaml`

## Can this PR be safely reverted / rolled back?
- [X] YES 💚
- [ ] NO ❌

## 🚨 User Impact 🚨
None

### CODE REVIEW

**Review by cgardens — APPROVED** [review]

File: airbyte-server/src/main/java/io/airbyte/server/apis/SourceApiController.java [hunk 1]
```diff
@@ -58,6 +58,7 @@ public CheckConnectionRead checkConnectionToSource(final SourceIdRequestBody sou
   @Post("/check_connection_for_update")
   @Secured({EDITOR})
   @SecuredWorkspace
+  @ExecuteOn(TaskExecutors.IO)
```
Comment (davinchia) [inline airbyte-server/src/main/java/io/airbyte/server/apis/SourceApiController.java:61]: thinking out loud: I think it'll be very useful to have a spot explaining what this does and was going to suggest in code comments. However, clicking into the annotation provides enough reader information. Perhaps the internal micronaut guide?

**Review by davinchia — APPROVED** [review]
Great!

I didn't look closely at the routes that have the IO executor annotation.  As long as we cover all routes that hit the default scheduler client and the event runner interface, we should be good!

File: airbyte-server/src/main/resources/application.yml [hunk 2]
```diff
@@ -1,6 +1,9 @@
 micronaut:
   application:
     name: airbyte-server
+  executors:
```
[the remark this answers is not in the document]
Comment (jdpgrailsdev) [reply airbyte-server/src/main/resources/application.yml:4]: That sets type of executor service that is used for the built-in "io" executor pool.  The [Micronaut documentation](https://docs.micronaut.io/latest/guide/#reactiveServer) has more details, but you can basically choose between `scheduled`, `cached`, `fixed` and `work stealing`, which map to the methods to create new pools in the JDK [Executors](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/concurrent/Executors.html) class.

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**